core/common/src/LocalTime.kt

```diff
@@ -0,0 +1,111 @@
+         * Examples of time in ISO-8601 format:
+         * - `18:43`
+         * - `18:43:00`
+         * - `18:43:00.500`
+         * - `18:43:00.123456789`
+         *
+         * @throws IllegalArgumentException if the text cannot be parsed or the boundaries of [LocalTime] are
+         * exceeded.
+         */
+        public fun parse(isoString: String): LocalTime
+
+        internal val MIN: LocalTime
+        internal val MAX: LocalTime
+    }
+
+    /**
+     * Constructs a [LocalTime] instance from the given time components.
+     *
+     * The supported ranges of components:
+     * - [hour] `0..23`
+     * - [minute] `0..59`
+     * - [second] `0..59`
+     * - [nanosecond] `0..999_999_999`
+     *
+     * @throws IllegalArgumentException if any parameter is out of range.
+     */
+    public constructor(hour: Int, minute: Int, second: Int = 0, nanosecond: Int = 0)
+
+    /** Returns the hour-of-day time component of this time value. */
+    public val hour: Int
+    /** Returns the minute-of-hour time component of this time value. */
+    public val minute: Int
+    /** Returns the second-of-minute time component of this time value. */
+    public val second: Int
+    /** Returns the nanosecond-of-second time component of this time value. */
+    public val nanosecond: Int
+
+    /**
+     * Compares `this` time value with the [other] time value.
+     * Returns zero if this value is equal to the other, a negative number if this value occurs earlier
+     * in the course of a typical day than the other, and a positive number if this value occurs
+     * later in the course of a typical day than the other.
+     *
+     * Note that, on days when there is a time overlap (for example, due to the daylight saving time
+     * transitions in autumn), a "lesser" wall-clock reading can, in fact, happen later than the
+     * "greater" one.
+     */
+    public override operator fun compareTo(other: LocalTime): Int
+
+    /**
+     * Converts this time value to the ISO-8601 string representation.
+     *
+     * @see LocalDateTime.parse
+     */
+    public override fun toString(): String
+}
+
+/**
+ * Converts this string representing a time value in ISO-8601 format to a [LocalTime] value.
+ *
+ * See [LocalTime.parse] for examples of time string representations.
+ *
+ * @throws IllegalArgumentException if the text cannot be parsed or the boundaries of [LocalTime] are exceeded.
+ */
+public fun String.toLocalTime(): LocalTime = LocalTime.parse(this)
+
+/**
+ * Combines this time's components with the specified date components into a [LocalDateTime] value.
+ */
+public fun LocalTime.atDate(year: Int, monthNumber: Int, dayOfMonth: Int = 0): LocalDateTime =
+    LocalDateTime(year, monthNumber, dayOfMonth, hour, minute, second, nanosecond)
+
+/**
+ * Combines this time's components with the specified date components into a [LocalDateTime] value.
+ */
+public fun LocalTime.atDate(year: Int, month: Month, dayOfMonth: Int = 0): LocalDateTime =
```

Comment:
I think we need a native speaker to determine whether `atDate` or `onDate` is the more idiomatic name in this occasion. https://forum.wordreference.com/threads/at-on-the-date.639568/ seems to imply that `atDate` is more formal and less common, but, frankly, I'm out of my depth here.

Reply:
I have to admit I'd be more partial to `atDate` because we already have `atTime`. Harmonizing those two names seems to me to be more important than being perfectly grammatically correct.

Reply:
I'd also go for symmetry meaning `atDate`

Reply:
I'm coming from Android/JVM and there's [`atDate`](https://docs.oracle.com/javase/8/docs/api/java/time/LocalTime.html#atDate-java.time.LocalDate-). Naturally, I'd always start by typing `.at` and see what's auto suggested.

Up until you raised the point, I never thought `onDate` does sound more natural.

Reply:
Ok, after some searching, it does seem like `atDate` is fairly common as well.